manifests/config.pp

```diff
@@ -37,6 +37,17 @@
     mode   => '0750',
   }
 
+  if $keycloak::tech_preview_features {
+    file { "${keycloak::install_base}/standalone/configuration/profile.properties":
+      ensure  => 'file',
+      owner   => $keycloak::user,
+      group   => $keycloak::group,
+      content => template('keycloak/profile.properties.erb'),
+      mode    => '0750',
+      require => File["${keycloak::install_base}/standalone/configuration"],
+    }
```

Comment:
Would be good to `notify` the Keycloak service class, `Class['keycloak::service']`. Also to allow undoing this change maybe set `ensure` to `$profile_properties_ensure` and in `init.pp` set the value based on presence of `tech_preview_features`.  This way if someone no longer set `tech_preview_features` the file will get deleted and Keycloak restarted.